src/openscm_runner/adapters/magicc7/_run_magicc_parallel.py

```diff
@@ -17,6 +19,43 @@
 LOGGER = logging.getLogger(__name__)
 
 
+class TemporaryDirectory:
+    """A temporary directory context manager which works like
+    tempfile.TemporaryDirectory but supports existing directories.
+
+    If instantiated without a directory, behaves exactly like
+    tempfile.TemporaryDirectory. If instantiated with an explicit directory
+    name, will use this directory instead and will *not* delete the directory
+    when exiting the context.
+    """
+    def __init__(self, tempdir: typing.Union[None, str] = None, **kwargs):
+        if tempdir is None:
+            self._td = tempfile.TemporaryDirectory(**kwargs)
+        else:
+            self._td = None
+            self._tempdir = tempdir
+
+    def __enter__(self) -> str:
+        if self._td is None:
+            return self._tempdir
+        else:
+            return self._td.__enter__()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self._td is not None:
+            self._td.__exit__(exc_type, exc_val, exc_tb)
+
+    def __repr__(self):
+        if self._td is None:
+            return f"<TemporaryDirectory {self._tempdir}>"
+        else:
+            return repr(self._td)
+
+    def cleanup(self):
+        if self._td is not None:
```

Comment:
Needed?  Otherwise will need a docstring

Reply:
It's part of the `tempfile.TemporaryDirectory` API, so I'd rather have it (less mental load in the future to know that the APIs are the same). Unfortunately, there is no docstring on the function in the stdlib that I could copy, so I added a basic docstring.